subsys/bluetooth/mesh/pb_adv.c

```diff
@@ -186,7 +212,12 @@ static void free_segments(void)
 			adv->ctx.busy = 0U;
 		}
 
-		bt_mesh_adv_unref(adv);
+		/* If the buffer is waiting to be sent, do not unref the buffer.
```

Comment:
Do we really need to send this segment if we aborting the transfer? Why we can't remove this buffer and reschedule the work?

Reply:
This is not always called from reset. This can be called at point where it has messages waiting on going out. This way the tx buffer is cleared for the next transfer, but the messages waiting on going out will be sent. There is several places the prov_clear_tx/free_segments are called and in some way it thinks it is finished as it was before. But now it still has messages going out because of the delay.

Reply:
But why the need to be sent? Looking at the logic above they are just dropped: https://github.com/zephyrproject-rtos/zephyr/blob/9074feabf6f827f6b37a69d77a5e13093799a905/subsys/bluetooth/mesh/pb_adv.c#L181-L189